packages/ui/Marquee/MarqueeDecorator.js

```diff
@@ -478,42 +498,46 @@ const MarqueeDecorator = hoc(defaultConfig, (config, Wrapped) => {
 
 			// TODO: absolute showing check (or assume that it won't be rendered if it isn't showing?)
 			if (node && this.distance == null && !this.props.marqueeDisabled) {
-				this.distance = this.calculateDistance(node);
+				const {width} = node.getBoundingClientRect();
+				const {scrollWidth} = node;
+
+				this.padding = this.getPadding(width);
+				this.distance = this.calculateDistance(width, scrollWidth, this.padding);
 				this.contentFits = !this.shouldAnimate(this.distance);
+			}
+		}
 
-				const overflow = this.calculateTextOverflow(this.distance);
-				this.setState(state => {
-					return state.overflow === overflow ? null : {overflow};
-				});
+		getPadding (width) {
+			const {marqueePadding} = this.props;
+
+			if (typeof marqueePadding === 'string') {
+				if (/^\d+(.\d+)?%$/.test(marqueePadding)) {
+					return width * Number.parseFloat(marqueePadding) / 100;
+				}
+
+				// warning for invalid string value;
+				return 0;
 			}
+
+			return scale(marqueePadding);
```

Comment:
Should the docs indicate the padding will be scaled?

Reply:
Updated